src/main/java/hudson/remoting/Engine.java

```diff
@@ -268,7 +268,7 @@ public synchronized void startEngine() throws IOException {
         } else {
             LOGGER.log(Level.INFO, "Using custom JAR Cache: {0}", jarCache);
         }
-        
+        LOGGER.log(Level.INFO, "Using agent version: {0}", Launcher.VERSION);
```

Comment:
It would be better to do it earlier, preferably at the beginning of the method. Work Dir and Jar Cache initialization may cause fatal failures, and it would be nice to show version info before that.

`workDirManager.setupLogging()` may impact the behavior of `LOGGER`, but it should be fine to print the message before it. In such case the message will just go to initial `java.util.logging` appenders like CLI.